app/controllers/questions_controller.rb

```diff
@@ -0,0 +1,48 @@
+class QuestionsController < ApplicationController
+  def show
+    return if !get_tenant
+
+    question = Question.includes(:answers, :user).where("id = ? and private = ?", params[:id], false).take
+    
+    output = question.as_json
+    if question
+      output["user_name"] = question.user.name
+
+      output["answers"] = []
+
+      question.answers.each do |answer|
+        answer_hash = answer.as_json
+        answer_hash["user_name"] = answer.user.name
+        output["answers"].push(answer_hash)
+      end
+    end
```

Comment:
Look into [ActiveModel::Serializers](https://github.com/rails-api/active_model_serializers) to define this json format outside of the controller